### PR TITLE
feat(i18n): add Arabic(ar-eg) translation for Egypt and middle east

### PR DESCRIPTION
**Pull request by HaniOurEdu**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 6799d49</samp>

This change adds a new locale file `ar-eg.ts` for the Arabic (Egypt) language and culture. It provides translations for the element-plus UI components, which are used to build Vue 3 applications. It enhances the internationalization and accessibility of the element-plus library.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 6799d49</samp>

* Add a new file for the Arabic (Egypt) locale ([link](https://github.com/element-plus/element-plus/pull/15125/files?diff=unified&w=0#diff-6745f4f51d05178ae382dfa8c7a104f8daea5a6b5f37920fadafda6739773d17R1-R147))

### CODE REVIEW

File: packages/locale/lang/ar-eg.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      goto: 'أذهب إلى',
+      pagesize: '/صفحة',
+      total: 'الكل {total}',
+      pageClassifier: '',
+      page: 'صفحة', // to be translated
+      prev: 'انتقل إلى الصفحة السابقة', // to be translated
+      next: 'انتقل إلى الصفحة التالية', // to be translated
+      currentPage: 'صفحة رقم {pager}', // to be translated
+      prevPages: 'صفحات {pager} السابقة', // to be translated
+      nextPages: 'صفحات {pager} التالية', // to be translated
```
Comment (plainheart) [inline packages/locale/lang/ar-eg.ts:97]: Please remove the comments as they have been translated.

File: packages/locale/lang/ar-eg.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    },
+    image: {
+      error: 'فشل',
+    },
+    pageHeader: {
+      title: 'عودة',
+    },
+    popconfirm: {
+      confirmButtonText: 'نعم', // to be translated
+      cancelButtonText: 'لا', // to be translated
```
Comment (plainheart) [inline packages/locale/lang/ar-eg.ts:144]: Remove the comments.

File: packages/locale/lang/ar-eg.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      error: 'مدخل غير صحيح',
+      close: 'أغلق هذا التبويب',
+    },
+    upload: {
+      deleteTip: 'اضغط ازالة لحذف المحتوى',
+      delete: 'حذف',
+      preview: 'عرض',
+      continue: 'إستمرار',
+    },
+    table: {
```
Comment (plainheart) [inline packages/locale/lang/ar-eg.ts:118]: Missing translations for the _**slider**_ component.
https://github.com/element-plus/element-plus/blob/main/packages/locale/lang/en.ts#L131-L135

**Review by plainheart — CHANGES_REQUESTED** [review]

File: packages/locale/lang/ar-eg.ts (hunk 3 above)
Comment (HaniOurEdu) [reply packages/locale/lang/ar-eg.ts:118]: Done

File: packages/locale/lang/ar-eg.ts (hunk 2 above)
Comment (HaniOurEdu) [reply packages/locale/lang/ar-eg.ts]: Done

File: packages/locale/lang/ar-eg.ts (hunk 1 above)
Comment (HaniOurEdu) [reply packages/locale/lang/ar-eg.ts]: Done

File: packages/locale/lang/ar-eg.ts (hunk 3 above)
Comment (plainheart) [reply packages/locale/lang/ar-eg.ts:118]: Do you forget to commit something? The slider component doesn't still get translated.

Comment (HaniOurEdu) [reply packages/locale/lang/ar-eg.ts:123]: Kindly check again

**Review by plainheart — APPROVED** [review]

**Pull request merged**